website/docs/plugin/framework/functions/errors.mdx

```diff
@@ -0,0 +1,149 @@
+returned from Terraform CLI at the end of command output:
+
+```console
+$ terraform plan
+# ... other plan output ...
+╷
+│ Error: Message
+│ 
+│   on example.tf line #:
+│    #: source configuration line
+╵
+```
+
+In the framework, you may encounter them in response structs or as returns from
+provider-defined function execution.:
+
+```go
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+```
+
+This is the most common form for `FuncError`: a single error whose text
+is the concatenated error text from one or more errors. This approach allows
+your provider to inform practitioners about all relevant errors at the same
+time, allowing practitioners to fix their configuration or environment more
+quickly. You should only concatenate a `FunctError` and never replace or
+remove information it.
+
+The next section will detail the concepts and typical behaviors of
+function error, while the final section will outline the typical methods for
+working with function error, using functionality from the available
+[`function` package](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function).
+
+## Function Error Concepts
+
+### Text
+
+`Text` is a practitioner-oriented description of the problem. This should
+contain sufficient detail to provide both general and more specific information
+regarding the issue. For example "Error executing function: foo can only contain
+letters, numbers, and digits."
+
+### FunctionArgument
+
+`FunctionArgument` identifies the specific function argument position that caused the
+error. Only errors that pertain to a function argument will include this information.
+
+### Working With Existing Function Errors
+
+#### ConcatFuncErrors
+
+When receiving `function.FuncError` from a function or method, such as
+`Run()`, these should typically be concatenated with the
+response function error for the method. This can be accomplished with the
+[`ConcatFuncErrors(in ...*FuncError)` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#ConcatFuncErrors).
+
+For example:
+
+```go
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+    var boolArg bool
+    var stringArg string
+
+    resp.Error = function.ConcatFuncErrors(resp.Error, req.Arguments.Get(ctx, &boolArg, &stringArg))
+
+    // ... other logic ...
+}
+```
+
+This method automatically ignores `nil` function errors.
+
+### Creating Function Errors
+
+To craft the message of a function error, it is recommended to use sufficient
+detail to convey both the cause of the error and as much contextual,
+troubleshooting, and next action information as possible. These details can
+use newlines for easier readability where necessary.
+
+#### NewFuncError
+
+When creating function errors that affect an entire data source, provider, or
+resource, and where a `function.FunctionError` is already available such as within
```

Comment:
> _When creating function errors that affect an entire data source, provider, or
resource_

I didn't quite understand what this sentence is referring to, can you expand on this?

Reply:
I've stripped that out.